src/main/scala/tilelink/AddressAdjuster.scala

```diff
@@ -7,203 +7,315 @@ import chisel3.util._
 import freechips.rocketchip.config._
 import freechips.rocketchip.diplomacy._
 
+case class TLSplitterNode(
+  clientFn:  TLClientPortParameters  => TLClientPortParameters  = { s => s },
+  managerFn: Seq[TLManagerPortParameters] => Seq[TLManagerPortParameters] = { s => s })(
+  implicit valName: ValName) extends TLCustomNode {
+
+  def resolveStar(iKnown: Int, oKnown: Int, iStars: Int, oStars: Int): (Int, Int) = {
+    require (iStars == 0, s"$context does not support :*=")
+    if (oStars > 0) {
+      require (2*iKnown >= oKnown, s"$context has $oKnown outputs and $iKnown inputs; cannot assign ${2*iKnown-oKnown} edges to resolve :=*")
+      (0, iKnown - oKnown)
+    } else {
+      require (oKnown == 2*iKnown, s"$context has $oKnown outputs and $iKnown inputs; but 2*$iKnown must be $oKnown")
+      (0, 0)
+    }
```

Comment:
I would be more strict; oStars must be either:
- 0 (your core is correct)
- 2: require oKnown==0 and output is (0, iKnown)